src/components/LanguageSelector/LanguageSelector.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from 'react'
+import classnames from 'classnames'
+import { Menu } from '../header/Menu/Menu'
+import { LanguageSelectorButton } from './LanguageSelectorButton'
+import { Button } from '../Button/Button'
+
+export type LanguageDefinition = {
+  label: string
+  label_en?: string
+  attr: string
+  on_click: string | (() => void)
+}
+
+type LanguageSelectorProps = {
+  label?: string
+  langs: LanguageDefinition[]
+  small?: boolean
+  className?: string
+}
+
+export const LanguageSelector = ({
+  label,
+  langs,
+  small,
+  className,
+  ...divProps
+}: LanguageSelectorProps &
+  JSX.IntrinsicElements['div']): React.ReactElement => {
+  const classes = classnames(
+    'usa-language-container',
+    {
+      [`usa-language--small`]: small !== undefined,
+    },
+    className
+  )
+
+  if (langs.length > 2) {
+    const [isOpen, setIsOpen] = useState(false)
+    const items = []
+    for (let i = 0; i < langs.length; i++) {
+      // eslint-disable-next-line security/detect-object-injection
+      const lang: LanguageDefinition = langs[i]
+      if (typeof lang.on_click === 'string') {
+        items.push(
+          <a href={lang.on_click} data-testid={lang.attr}>
+            <span lang={lang.attr}>
+              <strong>{lang.label}</strong>
+            </span>
+            {lang.label_en && ` (${lang.label_en})`}
+          </a>
+        )
+      } else {
+        items.push(
+          <Button
+            onClick={lang.on_click}
+            data-testid={lang.attr}
+            type="button"
+            unstyled>
+            <span lang={lang.attr}>
+              <strong>{lang.label}</strong>
+            </span>
+            {lang.label_en && ` (${lang.label_en})`}
+          </Button>
+        )
+      }
+    }
+    return (
+      <div className={classes} data-testid="languageSelector" {...divProps}>
+        <ul className="usa-language__primary usa-accordion">
+          <li className="usa-language__primary-item">
+            <LanguageSelectorButton
+              className={classes}
+              label={label || langs[0].label}
+              isOpen={isOpen}
+              onToggle={() => {
+                setIsOpen((prevIsOpen) => !prevIsOpen)
+              }}
+            />
+            <Menu
+              items={items}
+              isOpen={isOpen}
+              id="language-options"
+              type="language"
+            />
+          </li>
+        </ul>
+      </div>
+    )
+  } else {
+    if (label)
+      console.warn(
+        "LanguageSelector's label is not used when only two languages are available."
+      )
+    const [langIndex, setLangIndex] = useState(false)
+    const curLang = langs[Number(langIndex)]
+    const curLangNotEn =
+      curLang.attr && curLang.attr !== 'en' ? curLang.attr : undefined
```

Comment:
Bit of a nitpick: I'm wondering if you could write a test to cover when `curLangNotEn` is `undefined`?

Reply:
I'm actually going to slightly refactor this to be more language agnostic, will request a review on the changes...

Reply:
Ok, i've removed this bit so that the attr always shows (adds a few extraneous bytes to a page, but make this more language-flexible), and I renamed the input object to be called "lang_local" so that we don't assume this is used on a primarily `en` site